src/modules/number/index.ts

```
@@ -0,0 +1,190 @@
+   *
+   * @throws When options define `max < min`.
+   *
+   * @example
+   * faker.number.int() // 55422
+   * faker.number.int(100) // 52
+   * faker.number.int({ min: 1000000 }) // 1031433
+   * faker.number.int({ max: 100 }) // 42
+   *
+   * @since 8.0.0
+   */
+  int(options: number | { min?: number; max?: number } = {}): number {
+    if (typeof options === 'number') {
+      options = { max: options };
+    }
+
+    const { min = 0, max = min + 99999 } = options;
+
+    if (max === min) {
+      return min;
+    }
+
+    if (max < min) {
+      throw new FakerError(`Max ${max} should be greater than min ${min}.`);
+    }
+
+    return this.faker.mersenne.rand(max + 1, min);
+  }
+
+  /**
+   * Returns a single random floating-point number for the given precision or range and precision.
+   *
+   * @param options Precision or options object. Defaults to `{}`.
+   * @param options.min Lower bound for generated number. Defaults to `0`.
+   * @param options.max Upper bound for generated number. Defaults to `99999`.
+   * @param options.precision Number of digits after the decimal point. Negative numbers will be treated as `0`. Defaults to `2`.
+   *
+   * @example
+   * faker.number.float() // 51696.36
+   * faker.number.float(1) // 52023.2
+   * faker.number.float({ min: 1000000 }) // 212859.76
+   * faker.number.float({ max: 100 }) // 28.11
+   * faker.number.float({ precision: 1 }) // 84055.3
+   * faker.number.float({ min: 10, max: 100, precision: 3 }) // 57.315
+   *
+   * @since 8.0.0
+   */
+  float(
+    options: number | { min?: number; max?: number; precision?: number } = {}
+  ): number {
+    if (typeof options === 'number') {
+      options = {
+        precision: options,
+      };
+    }
+
+    const { precision = 2, min = 0, max = min + 99999 } = options;
+
+    if (max === min) {
+      return min;
+    }
+
+    if (max < min) {
+      throw new FakerError(`Max ${max} should be greater than min ${min}.`);
+    }
```

Do we need these checks? We call `this.int` later, that does these checks anyway.

---

I would say fail fast in this case

---

In this case I would say fail fast and so: yes